spec/models/user_spec.rb

```diff
@@ -0,0 +1,91 @@
+require "shoulda/matchers"
+require "rails_helper"
+
+describe User, type: :model do
+  it "has a valid factory" do
+    expect(build(:user)).to be_valid
+  end
+
+  let(:user){FactoryBot.create :user}
```

Comment:
`let` và `let!` có khác nhau như thế nào nhỉ

Reply:
@VuThiTranVan 
- Dạ đối với let thì đối tượng chỉ được khởi tạo duy nhất 1 lần khi được gọi tới trong 1 khối it (example)  và giá trị của đối tượng sẽ được cache trong suốt example đó. Giá trị của đối tượng trong mỗi example có thể khác nhau do giá trị của đối tượng khởi tạo sau khi lần đầu được gọi tới.

- Đối với let! thì sẽ được gọi ngầm trong khối block before. Đồng nghĩa với trước mỗi 1 example, giá trị của đối tượng trong let! sẽ luôn được khởi tạo sẵn.